website/project/model.py

```diff
@@ -2253,6 +2253,49 @@ def remove_contributors(self, contributors, auth=None, log=True, save=False):
 
         return True
 
+    def update_contributor(self, user, permission, visible, auth, save=False):
+        """ TODO: this method should be updated as a replacement for the main loop of
+        Node#manage_contributors. Right now there are redundancies, but to avoid major
+        feature creep this will not be included as this time.
+
+        Also checks to make sure unique admin is not removing own admin privilege.
+        """
+        assert self.has_permission(auth.user, ADMIN), "Only admins can modify contributor permissions"
```

Comment:
Might make sense to throw an OSF `PermissionsError` rather than an `AssertionError`. In general, use `assert` only when the condition is never expected to happen. In this case, it's possible that a dev might mistakenly call `update_contributor` with a non-admin user.